presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -168,15 +171,16 @@ private SqlQueryExecution(
             this.schedulerStats = requireNonNull(schedulerStats, "schedulerStats is null");
             this.statsCalculator = requireNonNull(statsCalculator, "statsCalculator is null");
             this.costCalculator = requireNonNull(costCalculator, "costCalculator is null");
+            this.preparedQuery = requireNonNull(preparedQuery, "preparedQuery is null");
+            this.accessControl = requireNonNull(accessControl, "accessControl is null");
+            this.queryExplainer = requireNonNull(queryExplainer, "queryExplainer is null");
+            this.warningCollector = requireNonNull(warningCollector, "warningCollector is null");
 
             checkArgument(scheduleSplitBatchSize > 0, "scheduleSplitBatchSize must be greater than 0");
             this.scheduleSplitBatchSize = scheduleSplitBatchSize;
 
             this.stateMachine = requireNonNull(stateMachine, "stateMachine is null");
 
-            // analyze query
-            this.analysis = analyze(preparedQuery, stateMachine, metadata, accessControl, sqlParser, queryExplainer, warningCollector);
-
```

Comment:
Performing analyze in constructor causes QueryCompletedEvent to be lost in the case of an EXPLAIN failing. The member variable is only accessed below in doPlanQuery, so storing it as a member variable is not necessary.

Reply:
Not sure how to write a unit test for this. It could be done pretty easily on top of https://github.com/prestosql/presto/pull/2334 if a new connector method is added for validation.

Reply:
Never mind - I figured out an easy way and added a test for this.